primitives/tf_encrypted/primitives/paillier/primitives.py

```diff
@@ -0,0 +1,128 @@
+from typing import Optional
+
+import numpy as np
+import tensorflow as tf
+
+import tf_big
+
+tf_big.set_secure_default(True)
+
+
+class EncryptionKey:
+    def __init__(self, n):
+        n = tf_big.convert_to_tensor(n)
+
+        self.n = n
+        self.nn = n * n
+        self.g = 1 + n
+
+
+class DecryptionKey:
+    def __init__(self, p, q):
+        p = tf_big.convert_to_tensor(p)
+        q = tf_big.convert_to_tensor(q)
+
+        n = p * q
+        self.n = n
+        self.nn = n * n
+        self.g = 1 + n
+
+        order_of_n = (p - 1) * (q - 1)
+        self.d1 = order_of_n
+        self.d2 = tf_big.inv(order_of_n, n)
+        self.e = tf_big.inv(n, order_of_n)
+
+
+def gen_keypair(bitlength=2048):
+    p, q, n = tf_big.random_rsa_modulus(bitlength=bitlength)
+    return EncryptionKey(n), DecryptionKey(p, q)
+
+
+class Randomness:
+    def __init__(self, raw_randomness):
+        self.raw = tf_big.convert_to_tensor(raw_randomness)
+
+
+def gen_randomness(ek, shape, dtype: tf.DType = tf.uint8):
+    return Randomness(tf_big.random_uniform(shape=shape, maxval=ek.n))
+
+
+class Ciphertext:
+    def __init__(self, ek: EncryptionKey, raw_ciphertext):
+        self.ek = ek
+        self.raw = tf_big.convert_to_tensor(raw_ciphertext)
```

Comment:
do we necessarily want to have `self.raw` be a BigTensor or it can be a normal tensor (e.g tf.string)? if we ask `encrypt` to return c as tf.string, `tf_big.convert_to_tensor` will turn it back into a tf.variant.